rest_api/test/test_rest_api.py

```diff
@@ -366,6 +368,57 @@ def test_query_with_bool_in_params(client):
         assert response_json["answers"] == []
 
 
+def test_query_with_embeddings(client):
+    with mock.patch("rest_api.controller.search.query_pipeline") as mocked_pipeline:
+        # `run` must return a dictionary containing a `query` key
+        mocked_pipeline.run.return_value = {
+            "query": TEST_QUERY,
+            "documents": [
+                Document(
+                    content="test",
+                    content_type="text",
+                    score=0.9,
+                    meta={"test_key": "test_value"},
+                    embedding=np.array([0.1, 0.2, 0.3]),
+                )
+            ],
+        }
+        response = client.post(url="/query", json={"query": TEST_QUERY})
+        assert 200 == response.status_code
+        assert len(response.json()["documents"]) == 1
+        assert response.json()["documents"][0]["content"] == "test"
+        assert response.json()["documents"][0]["content_type"] == "text"
+        assert response.json()["documents"][0]["embedding"] == [0.1, 0.2, 0.3]
```

Comment:
You might make it a bit more readable by doing:

```suggestion
        assert response.json() == [{
            "query": TEST_QUERY,
            "documents": {
                "content": "test",
                "content_type": "test",
                "embedding": [0.1, 0.2, 0.3],
                "score": 0.9,
                "meta": {"test_key": "test_value"},
            }
        }]
```

(same in other tests below)

Reply:
There would be additional params inside the response we really don't want to test, e.g. score, document id, etc...